lib/ramble/ramble/reports.py

```diff
@@ -0,0 +1,989 @@
+        # TODO: how to handle success? Repeats make this tricky because some children might have
+        # status=success but parent might have status=failed
+        # if exp['RAMBLE_STATUS'] != 'SUCCESS' or exp['name'] in skip_exps:
+
+        # TODO: disable skip_exps in favor of a more explicit is_child or is_parent check
+        if exp["name"] in skip_exps or is_repeat_child(exp):
+            logger.debug(f"Skipping import of experiment {exp['name']}")
+            continue
+        # FIXME: is this wise? We need to flesh out the CUJ around success/fail
+        elif exp["RAMBLE_STATUS"] != "SUCCESS":
+            continue
+        else:
+            logger.debug(f"Importing experiment {exp['name']}")
+            # For repeat experiments, use summary stats from base exp and skip repeats
+            # Repeats are sequenced after base exp
+
+            if exp.get("N_REPEATS", 0) > 0:
+                # Generate repeat experiment names in order to skip them explicitly
+                exp_name = exp["name"]
+                for n in range(1, exp["N_REPEATS"] + 1):
+                    if ".chain" in exp_name:
+                        insert_idx = exp_name.index(".chain")
+                        repeat_exp_name = exp_name[:insert_idx] + f".{n}" + exp_name[insert_idx:]
+                        skip_exps.append(repeat_exp_name)
+                    else:
+                        skip_exps.append(exp_name + f".{n}")
+
+            for context in exp["CONTEXTS"]:
+                for fom in context["foms"]:
+                    # Expand to one row/FOM/context w/ a copy of the experiment vars and metadata
+                    exp_copy = copy.deepcopy(exp)
+
+                    # Remove context dict and add the current FOM values
+                    exp_copy.pop("CONTEXTS")
+                    exp_copy[ReportVars.CONTEXT.value] = context["name"]
+                    for name, val in fom.items():
+                        if name in _FOM_DICT_MAPPING.keys():
+                            exp_copy[_FOM_DICT_MAPPING[name]] = val
+                        elif name == "fom_type":
+                            exp_copy["fom_type"] = FomType.from_str(fom["fom_type"]["name"])
+                            exp_copy[ReportVars.BETTER_DIRECTION.value] = BetterDirection.from_str(
+                                fom["fom_type"][ReportVars.BETTER_DIRECTION.value]
+                            )
+
+                        # older data exports may not have fom_type stored
+                        if "fom_type" not in exp_copy:
+                            exp_copy["fom_type"] = FomType.UNDEFINED
+                            exp_copy[ReportVars.BETTER_DIRECTION.value] = (
+                                BetterDirection.INDETERMINATE
+                            )
+
+                    # Exclude vars that aren't needed for analysis, mainly paths and commands
+                    dir_regex = r"_dir$"
```

Comment:
Maybe these definitions should be higher, since they are static?

Reply:
I think since they are only used here, it's OK to keep them local. I think if we need them elsewhere too that is the time to move them?